src/components/Settings/Settings.js

```diff
@@ -30,18 +42,122 @@ export default class Settings extends React.Component {
             <option value="TestNet">TestNet</option>
             <option value="CozNet">CozNet</option>
             <option value="nOSLocal">nOS Local</option>
+
+            {
+              this.props.allNetworks.map((network) => {
+                return (
+                  <option
+                    key={network.neoscan}
+                    value={network.neoscan}
+                  >
+                    {network.name}
+                  </option>
+                );
+              })
+            }
           </select>
         </label>
+        <div className={styles.buttonContainer}>
+          Custom Neoscan Url: {this.props.currentNetwork}
+        </div>
+        {this.renderButtons()}
         {this.state.saved && <Saved /> }
       </div>
     );
   }
 
+  renderButtons = () => {
+    return (
+      <div>
+        <Button onClick={this.handleAddNewNetwork}>
+          {'Add custom network configuration'}
+        </Button>
+        <div className={styles.divider} />
+        <Button onClick={this.handleClearNetwork}>
+          {'Clear custom network configurations'}
+        </Button>
+      </div>
+    );
+  }
+
+  handleClearNetwork = () => {
+    this.props.clearNetworks();
+    this.props.setCurrentNetwork('TestNet');
+    this.props.alert('All custom network configurations cleared');
+  }
+
+  handleAddNewNetwork = () => {
+    this.props.confirm((
+      <div>
+        <Input
+          id="networkName"
+          type="text"
+          label="Network name"
+          placeholder="Network name"
+          onChange={this.handleChangeNetworkName}
+        />
+        <Input
+          id="networkURL"
+          type="text"
+          label="Network URL"
+          placeholder="Network URL"
+          onChange={this.handleChangeNetworkUrl}
+        />
+      </div>
+    ), {
+      title: 'New network configuration',
+      onConfirm: this.handleConfirmAddNetwork,
+      onCancel: noop
+    });
+  }
+
+  handleConfirmAddNetwork = () => {
+    const network = this.props.allNetworks.find((element) => {
+      return element.neoscan === this.props.networkUrl;
+    });
+
+    if (network) {
+      this.props.alert('Error: A network configuration with that Url already exist');
```

Comment:
🔤 Capitalize "URL" and end sentence with a period.  (The capitalization may be a moot point based upon my prior comment.)